src/plugins/usage_collection/server/usage_counters/saved_objects.ts

```diff
@@ -34,19 +35,43 @@ export interface UsageCountersSavedObjectAttributes {
 export type UsageCountersSavedObject = SavedObject<UsageCountersSavedObjectAttributes>;
 
 /** The Saved Objects type for Usage Counters **/
-export const USAGE_COUNTERS_SAVED_OBJECT_TYPE = 'usage-counters';
+export const SERVER_COUNTERS_SAVED_OBJECT_TYPE = 'server-counters';
+export const UI_COUNTERS_SAVED_OBJECT_TYPE = 'ui-counters';
+export type UsageCounterSavedObjectType = 'server-counters' | 'ui-counters';
+export const USAGE_COUNTERS_SAVED_OBJECT_TYPES = [
+  SERVER_COUNTERS_SAVED_OBJECT_TYPE,
+  UI_COUNTERS_SAVED_OBJECT_TYPE,
+];
 
-export const registerUsageCountersSavedObjectType = (
+export const registerUsageCountersSavedObjectTypes = (
   savedObjectsSetup: SavedObjectsServiceSetup
 ) => {
   savedObjectsSetup.registerType({
-    name: USAGE_COUNTERS_SAVED_OBJECT_TYPE,
+    indexPattern: USAGE_COUNTERS_SAVED_OBJECT_INDEX,
+    name: SERVER_COUNTERS_SAVED_OBJECT_TYPE,
     hidden: false,
-    namespaceType: 'agnostic',
+    namespaceType: 'single',
     mappings: {
       dynamic: false,
       properties: {
         domainId: { type: 'keyword' },
+        counterName: { type: 'keyword' },
+        counterType: { type: 'keyword' },
+      },
```

Comment:
I think that we need to increase the mappings here: 
If we're planning to query by date, `fromTimestamp` should be stored (instead of calculated in the collector). Right?

Reply:
Related to https://github.com/elastic/kibana/pull/187064/files#r1658379222